frontend/packages/ceph-storage-plugin/src/components/block-pool/body.tsx

```diff
@@ -0,0 +1,265 @@
+import { CEPH_STORAGE_NAMESPACE, OCS_DEVICE_REPLICA } from '../../constants/index';
+import { checkArbiterCluster } from '../../utils/common';
+import { PROGRESS_STATUS, ProgressStatusProps } from '../../utils/storage-pool';
+import { POOL_STATE, POOL_PROGRESS } from '../../constants/storage-pool-const';
+import {
+  BlockPoolState,
+  BlockPoolAction,
+  BlockPoolActionType,
+  getErrorMessage,
+} from '../../utils/block-pool';
+import { GUARDED_FEATURES } from '../../features';
+
+import './body.scss';
+
+export const BlockPoolStatus: React.FC<BlockPoolStatusProps> = ({ status, name, error = '' }) => {
+  const { t } = useTranslation();
+  const statusObj: ProgressStatusProps = PROGRESS_STATUS(t).find((state) => state.name === status);
+
+  return (
+    <>
+      <EmptyState>
+        <EmptyStateIcon icon={statusObj.icon} className={statusObj.className} />
+        <EmptyStateBody data-test="empty-state-body">
+          {error ? getErrorMessage(error) : statusObj.desc.replace('{name}', name)}
+        </EmptyStateBody>
+      </EmptyState>
+    </>
+  );
+};
+
+export type BlockPoolStatusProps = {
+  status: string;
+  name?: string;
+  error?: string;
+};
+
+export const BlockPoolBody = (props: BlockPoolBodyPros) => {
+  const { cephCluster, state, dispatch, showPoolStatus } = props;
+  const { t } = useTranslation();
+
+  const isPoolManagementSupported = useFlag(GUARDED_FEATURES.OCS_POOL_MANAGEMENT);
+  const [storageCluster, storageClusterLoaded, storageClusterLoadError] = useK8sGet<
+    ListKind<StorageClusterKind>
+  >(OCSServiceModel, null, CEPH_STORAGE_NAMESPACE);
+
+  const [isReplicaOpen, setReplicaOpen] = React.useState(false);
+  const [isVolumeTypeOpen, setVolumeTypeOpen] = React.useState(false);
+  const [availableDeviceClasses, setAvailableDeviceClasses] = React.useState([]);
+
+  // Volume Type
+  const deviceClasses = cephCluster?.status?.storage?.deviceClasses ?? [];
+
+  React.useEffect(() => {
+    if (deviceClasses.length && isPoolManagementSupported) {
+      const setVolumeType = (volumeType: string) =>
+        dispatch({ type: BlockPoolActionType.SET_POOL_VOLUME_TYPE, payload: volumeType });
+
+      if (state.volumeType === '') {
+        // Set default value
+        const ssdDeviceClass =
+          deviceClasses.find((deviceClass) => deviceClass.name === 'ssd') || {};
+        Object.keys(ssdDeviceClass).length
+          ? setVolumeType('ssd')
+          : setVolumeType(deviceClasses[0].name);
+      }
+
+      // Volume Type dropdown
+      setAvailableDeviceClasses(
+        deviceClasses.map((device) => {
+          return (
+            <DropdownItem
+              key={`device-${device?.name}`}
+              component="button"
+              id={device?.name}
+              data-test={device?.name}
+              onClick={(e) => setVolumeType(e.currentTarget.id)}
+            >
+              {device?.name.toUpperCase()}
+            </DropdownItem>
+          );
+        }),
+      );
+    }
```

Comment:
This still does not updates if deviceClasses are changed.
Instead of setting it once for volumeType === '' , you must set whenever the list of volumes changes.
```
  const deviceClasses = cephCluster?.status?.storage?.deviceClasses ?? [];

  if (
    isPoolManagementSupported &&
    deviceClasses.length &&
    !_.isEqual(deviceClasses, availableDeviceClasses)
  ) {
    setAvailableDeviceClasses(deviceClasses);
    const ssdDeviceClass: CephDeviceClass = availableDeviceClasses.find(
      (deviceClass) => deviceClass.name === 'ssd',
    );
    ssdDeviceClass?.name
      ? setVolumeType(ssdDeviceClass.name)
      : setVolumeType(availableDeviceClasses[0].name);
  }
  ```

Reply:
Also you can set a state for deviceClasses and then create dropdowns.
```
  const volumeTypeDropdownItems = availableDeviceClasses.map((device) => {
    return (
      <DropdownItem
        key={`device-${device?.name}`}
        component="button"
        id={device?.name}
        data-test={device?.name}
        onClick={(e) => setVolumeType(e.currentTarget.id)}
      >
        {device?.name.toUpperCase()}
      </DropdownItem>
    );
  });
  ```

Reply:
For `volumeSetType` you can useCallback
```
  const setVolumeType = React.useCallback(
    (volumeType: string) =>
      dispatch({ type: BlockPoolActionType.SET_POOL_VOLUME_TYPE, payload: volumeType }),
    [dispatch],
  );
  ```

Reply:
let me check this, i not aware this callback usage inside useEffect

Reply:
Its not supposed to be inside :)

Reply:
It will change, no need to do the above change:

if device class change happen only useEffect is executed:

In code i am recreating drop down every time, the if condition if (state.volumeType === '')  is  just for display SSD as default
```
setAvailableDeviceClasses(
        deviceClasses.map((device) => {
          return (
            <DropdownItem
              key={`device-${device?.name}`}
              component="button"
              id={device?.name}
              data-test={device?.name}
              onClick={(e) => setVolumeType(e.currentTarget.id)}
            >
              {device?.name.toUpperCase()}
            </DropdownItem>
          );
        }),
      );
```

Reply:
I am pointing here the case when you dont have any device classes.
You fetched ceph cluster -> get Device classes -> no device classes - volumeType -> set to `undefined` as per current logic.
Then after some time suppose some device classes are discovered then it will never set the default values because `volumeType === ' '` will be falsy since `undefined !== ' '`

Another thing setting device classes in state makes more sense rather than dropdown.
We generally dont put rendering logic in state.

Reply:
initial value of the volume type is '', and updating volume type in 3 different places 
   first two is  inside if (state.volumeType === '')
   third one is onClick={(e) => setVolumeType(e.currentTarget.id)}, there is nothing to click in drop down so no chance of undefined.

Reply:
My perception remains same but wont block PR on that.